server/public/plugin/client_rpc.go

```diff
@@ -358,9 +359,39 @@ func init() {
 	hookNameToId["ServeHTTP"] = ServeHTTPID
 }
 
+// Using a subset of http.Request prevents a known incompatibility when decoding Go v1.23+ gob-encoded x509.Certificate
+// structs from Go v1.22 compiled plugins. These come from http.Request.TLS field (*tls.ConnectionState).
+type HTTPRequest struct {
+	Method     string
+	URL        *url.URL
+	Proto      string
+	ProtoMajor int
+	ProtoMinor int
+	Header     http.Header
+	Host       string
+	RemoteAddr string
+	RequestURI string
+	Body       io.ReadCloser
+}
+
+func (r *HTTPRequest) GetHTTPRequest() *http.Request {
```

Comment:
nit: I wonder if naming the struct, `HTTPRequestSubset` or `SimplifiedHTTPRequest` might help distinguish `HTTPRequest` from `http.Request`? (0/5, honestly)